### ISSUE TITLE
Drawing triangle throws pyglet.gl.lib.GLException

### ISSUE THREAD

**Issue by LvanderGoten** [pvcraven/arcade#277]
## Bug Report

Drawing a triangle throws a **pyglet.gl.lib.GLException** while drawing other primitives works flawlessly.
### Actual behavior:

> Traceback (most recent call last):
>   File "/home/lennartv/PycharmProjects/BilliardAI/test.py", line 20, in <module>
>     arcade.run()
>   File "/usr/lib/python3.7/site-packages/arcade/window_commands.py", line 223, in run
>     pyglet.app.run()
>   File "/usr/lib/python3.7/site-packages/pyglet/app/__init__.py", line 138, in run
>     event_loop.run()
>   File "/usr/lib/python3.7/site-packages/pyglet/app/base.py", line 142, in run
>     self._run()
>   File "/usr/lib/python3.7/site-packages/pyglet/app/base.py", line 154, in _run
>     timeout = self.idle()
>   File "/usr/lib/python3.7/site-packages/pyglet/app/base.py", line 281, in idle
>     window.dispatch_event('on_draw')
>   File "/usr/lib/python3.7/site-packages/pyglet/window/__init__.py", line 1232, in dispatch_event
>     if EventDispatcher.dispatch_event(self, *args) != False:
>   File "/usr/lib/python3.7/site-packages/pyglet/event.py", line 367, in dispatch_event
>     if getattr(self, event_type)(*args):
>   File "/home/lennartv/PycharmProjects/BilliardAI/test.py", line 11, in on_draw
>     x3=200, y3=200, color=arcade.color.WHITE)
>   File "/usr/lib/python3.7/site-packages/arcade/draw_commands.py", line 912, in draw_triangle_filled
>     draw_polygon_filled(point_list, color)
>   File "/usr/lib/python3.7/site-packages/arcade/draw_commands.py", line 859, in draw_polygon_filled
>     _generic_draw_line_strip(point_list, color, 1, gl.GL_POLYGON)
>   File "/usr/lib/python3.7/site-packages/arcade/draw_commands.py", line 734, in _generic_draw_line_strip
>     vao.render(mode=mode)
>   File "/usr/lib/python3.7/site-packages/arcade/shader.py", line 498, in render
>     glDrawArraysInstanced(mode, 0, self.num_vertices, instances)
>   File "/usr/lib/python3.7/site-packages/pyglet/gl/lib.py", line 105, in errcheck
>     raise GLException(msg)
> pyglet.gl.lib.GLException: b'invalid enumerant'

### Steps to reproduce/example code:

```python
import arcade


class Game(arcade.Window):
    def __init__(self, screen_width, screen_height):
        super().__init__(width=screen_width, height=screen_height, title="Test")

    def on_draw(self):
        """ Throws pyglet.gl.lib.GLException """
        arcade.draw_triangle_filled(x1=50, y1=50,
                                    x2=100, y2=100,
                                    x3=200, y3=200, color=arcade.color.WHITE)
        """ This works however """
        # arcade.draw_rectangle_filled(center_x=50, center_y=50,
        #                              width=20, height=20, color=arcade.color.WHITE)

    def on_update(self, delta_time):
        pass


game = Game(screen_width=500, screen_height=500)
arcade.run()
```
## Specifics

- Arcade: 2.0.0a1
- Python 3.7.1
- OS: Arch Linux 4.18.16-arch1-1-ARCH
- OpenGL: 3.0 Mesa 18.2.4

**Comment by pvcraven** [comment]
Thank you so much for the report! Your code isn't crashing under Windows. It'll take me a bit to try it under Linux.

I note that there's a missing `start_render` and the triangle isn't a triangle. Does this code run, or does it also crash?

`import arcade


class Game(arcade.Window):
    def __init__(self, screen_width, screen_height):
        super().__init__(width=screen_width, height=screen_height, title="Test")

    def on_draw(self):
        arcade.start_render()
        """ Throws pyglet.gl.lib.GLException """
        arcade.draw_triangle_filled(x1=50, y1=50,
                                    x2=50, y2=200,
                                    x3=200, y3=200, color=arcade.color.WHITE)
        """ This works however """
        # arcade.draw_rectangle_filled(center_x=50, center_y=50,
        #                              width=20, height=20, color=arcade.color.WHITE)

    def on_update(self, delta_time):
        pass


game = Game(screen_width=500, screen_height=500)
arcade.run()`

**Comment by LvanderGoten** [comment]
Hi!

I get the same error
```
Traceback (most recent call last):
  File "/home/lennartv/PycharmProjects/BilliardAI/bug.py", line 23, in <module>
    arcade.run()
  File "/usr/lib/python3.7/site-packages/arcade/window_commands.py", line 223, in run
    pyglet.app.run()
  File "/usr/lib/python3.7/site-packages/pyglet/app/__init__.py", line 138, in run
    event_loop.run()
  File "/usr/lib/python3.7/site-packages/pyglet/app/base.py", line 142, in run
    self._run()
  File "/usr/lib/python3.7/site-packages/pyglet/app/base.py", line 154, in _run
    timeout = self.idle()
  File "/usr/lib/python3.7/site-packages/pyglet/app/base.py", line 281, in idle
    window.dispatch_event('on_draw')
  File "/usr/lib/python3.7/site-packages/pyglet/window/__init__.py", line 1232, in dispatch_event
    if EventDispatcher.dispatch_event(self, *args) != False:
  File "/usr/lib/python3.7/site-packages/pyglet/event.py", line 367, in dispatch_event
    if getattr(self, event_type)(*args):
  File "/home/lennartv/PycharmProjects/BilliardAI/bug.py", line 13, in on_draw
    x3=200, y3=200, color=arcade.color.WHITE)
  File "/usr/lib/python3.7/site-packages/arcade/draw_commands.py", line 912, in draw_triangle_filled
    draw_polygon_filled(point_list, color)
  File "/usr/lib/python3.7/site-packages/arcade/draw_commands.py", line 859, in draw_polygon_filled
    _generic_draw_line_strip(point_list, color, 1, gl.GL_POLYGON)
  File "/usr/lib/python3.7/site-packages/arcade/draw_commands.py", line 734, in _generic_draw_line_strip
    vao.render(mode=mode)
  File "/usr/lib/python3.7/site-packages/arcade/shader.py", line 498, in render
    glDrawArraysInstanced(mode, 0, self.num_vertices, instances)
  File "/usr/lib/python3.7/site-packages/pyglet/gl/lib.py", line 105, in errcheck
    raise GLException(msg)
pyglet.gl.lib.GLException: b'invalid enumerant'
```

**Comment by pvcraven** [comment]
Running Linux as well?

**Comment by LvanderGoten** [comment]
Yes, I ran it on the same system.

**Comment by pvcraven** [comment]
Calling `glDrawArraysInstanced` with `GL_TRIANGLES` and `GL_QUADS` works with Linux, but not `GL_POLYGON`. Not sure why, as this seems to work on Windows ok.

**Comment by pvcraven** [comment]
I uploaded release 2.0.0a2 which should allow triangle drawing. It has an issue with drawing polygons on Linux still.

**Comment by dangillet** [comment]
You should use neither GL_QUADS nor GL_POLYGONS as its deprecated since OpenGL 3. Some implementations might still support it, but as seen here it’s not supported everywhere. 

All quads and polygons can easily be tessalated into triangle strips.

**Comment by pvcraven** [comment]
Ok, this is done for the next release.

**Comment by pvcraven** [comment]
Closed with 2.0 release.

**Comment by thecubeschoolhub** [comment]
Hi there!
I'm experiencing a similar issue.

I asked a very similar question on stack overflow [here](https://stackoverflow.com/questions/55404539/python-3-6-4-arcade-module-window-errors-pyglet-gl-lib-glexception-binvalid)

I'm running the exact same arcade code on two computers. Works on one but not on the other.
The difference i discovered:
One of the computers has an nvidia graphics card and the other doesn't (just regular intel HD R graphics). Is there a way to get the arcade module working with a special OpenGL install on the intel HD R computer?
